src/include/misc.h

```diff
@@ -312,6 +312,14 @@ __wt_string_match(const char *str, const char *bytes, size_t len)
     return (len > 0 && (strlen(str) == len) && (strncmp(str, bytes, len) == 0));
 }
 
+/*
+ * This is a faster version of WT_STRING_MATCH, and should be used where performance matters. bytes
+ * or str should never be NULL.
+ */
+#define WT_FAST_STRING_MATCH(str, bytes, len)                                                   \
```

Comment:
In the case `len==0` even assuming `str` and `bytes` are both valid pointers to readable memory; this will make the invalid comparison `str[0]==bytes[0]` potentially resulting in an evaluation of false.

```suggestion
#define WT_FAST_STRING_MATCH(str, bytes, len)                                                   \
 ((len==0) || (((const char *)(str))[0] == ((const char *)(bytes))[0] && strncmp(str, bytes, len) == 0 && \
      (str)[len] == '\0'))
```

Reply:
@marcbutler - In the limited uses of this macro, I think at least the first string is non-zero length, so it shouldn't trigger the invalid case.  I'll add an assert and comment.

Reply:
In the cases I'm interested in, the first arg is known to have non-zero length.  I made a WT_RESTRICTED_STRING_MATCH that uses an assert to check the constraints.  Let me know what you think.